common/src/main/scala/com/gu/media/Permissions.scala

```diff
@@ -41,3 +51,11 @@ class MediaAtomMakerPermissionsProvider(stage: String, credsProvider: AWSCredent
     }
   }
 }
+
+object PermissionsUploadHelper {
+  def permissionAndIntentMatch(permission: Permissions, selfHosted: Boolean) : Boolean = {
```

Comment:
I'd move this into UploadController directly or rename it to be a bit clearer (something like `canPerformUpload`).

Thinking about it, I wonder if we can refactor the code in UploadController to have two clear routes from the entry point, one for YouTube and another for self hosted. Ideally we wouldn't need this method which handles both routes.